### PR TITLE
fix: [SRM-13791]: fixed validations and some logic for downtime

### PR DESCRIPTION
**Pull request by karansaraswat19**
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by karansaraswat19** [comment]
retrigger jest

**Comment by karansaraswat19** [comment]
retrigger cypress-rest

**Comment by karansaraswat19** [comment]
retrigger jest

File: src/modules/85-cv/pages/slos/SLODowntimePage/components/DowntimeList/DowntimeList.utils.ts [hunk 1]
```diff
@@ -5,14 +5,80 @@
...
 import { DowntimeStatus } from '../../SLODowntimePage.types'
 
 export const getIsSetPreviousPage = (pageIndex: number, pageItemCount: number): boolean => {
   return Boolean(pageIndex) && pageItemCount === 1
 }
 
+export const getDowntimeWindowInfo = (downtime: DowntimeListView, getString: UseStringsReturn['getString']) => {
+  const { type = DowntimeWindowToggleViews.ONE_TIME } = downtime?.spec || {}
+  const { startTime = 1, timezone = 'Asia/Calcutta' } = downtime?.spec?.spec || {}
+
```
Comment (praneshg239) [inline src/modules/85-cv/pages/slos/SLODowntimePage/components/DowntimeList/DowntimeList.utils.ts:32]: Please refactor this function and make it readable

File: src/modules/85-cv/pages/slos/components/CVCreateDowntime/__tests__/CVCreateDowntime.test.tsx [hunk 2]
```diff
@@ -67,6 +68,21 @@ describe('CVCreateDowntime', () => {
     })
 
     expect(container).toMatchSnapshot()
+
+    act(() => {
+      userEvent.click(getByText('next'))
+    })
+
+    // await waitFor(() => {
```
Comment (praneshg239) [inline src/modules/85-cv/pages/slos/components/CVCreateDowntime/__tests__/CVCreateDowntime.test.tsx:76]: remove the commented code if not used

File: src/modules/85-cv/pages/slos/components/CVCreateDowntime/components/CreateDowntimeForm/CreateDowntimeForm.utils.ts [hunk 3]
```diff
@@ -77,7 +77,7 @@ export const validateSelectDowntimeWindowSection = (formikProps: FormikProps<Dow
       formikProps.setFieldTouched(DowntimeFormFields.END_TIME, true)
 
       const { endTime } = formikProps.values
-      if (!endTime || moment(endTime).isBefore(moment(startTime))) {
+      if (!endTime || moment(endTime).isBefore(moment(startTime)) || moment(endTime) > moment(startTime).add(3, 'y')) {
```
Comment (deepesh-ui) [inline src/modules/85-cv/pages/slos/components/CVCreateDowntime/components/CreateDowntimeForm/CreateDowntimeForm.utils.ts:80]: move this logic to a variable 
` moment(endTime) > moment(startTime).add(3, 'y')`

**Review by praneshg239 — APPROVED** [review]

**Comment by karansaraswat19** [comment]
retrigger jset

**Comment by karansaraswat19** [comment]
retrigger jest

**Comment by karansaraswat19** [comment]
retrigger jest

**Comment by karansaraswat19** [comment]
retrigger jest

File: src/modules/85-cv/pages/slos/SLODowntimePage/components/DowntimeList/DowntimeList.utils.ts [hunk 4]
```diff
@@ -5,27 +5,91 @@
...
+    if (oneTimeDowntimeType === EndTimeMode.DURATION) {
+      const { durationValue = 30, durationType = 'Minutes' } =
+        (onetimeDowntimeSpec?.spec as OnetimeDurationBasedSpec)?.downtimeDuration || {}
+      _endTime = moment(_startTime)
+        .add(durationValue, durationType.toLowerCase() as any)
+        .format('lll')
+    } else {
+      const { endTime = 1 } = (onetimeDowntimeSpec?.spec as OnetimeEndTimeBasedSpec) || {}
+      _endTime = getFormattedTime({
```
Comment (harshilbits) [inline src/modules/85-cv/pages/slos/SLODowntimePage/components/DowntimeList/DowntimeList.utils.ts:50]: use camelCase only.

File: src/modules/85-cv/pages/slos/SLODowntimePage/components/DowntimeList/DowntimeList.utils.ts [hunk 5]
```diff
@@ -5,27 +5,91 @@
...
+  let timeFrame = null
+  let downtimeType = null
+
+  if (type === DowntimeWindowToggleViews.ONE_TIME) {
+    const onetimeDowntimeSpec = downtime?.spec?.spec as OnetimeDowntimeSpec
+    const { type: oneTimeDowntimeType = EndTimeMode.DURATION } = onetimeDowntimeSpec || {}
+    const _startTime = getFormattedTime({ time: startTime, timezone, format: 'lll' })
+    let _endTime = null
+
+    if (oneTimeDowntimeType === EndTimeMode.DURATION) {
```
Comment (harshilbits) [inline src/modules/85-cv/pages/slos/SLODowntimePage/components/DowntimeList/DowntimeList.utils.ts:42]: Split this into small readable functions with meaningful name

File: src/modules/85-cv/pages/slos/SLODowntimePage/components/DowntimeList/DowntimeList.utils.ts [hunk 6]
```diff
@@ -5,27 +5,91 @@
...
+      const { durationValue = 30, durationType = 'Minutes' } =
```
Comment (harshilbits) [inline src/modules/85-cv/pages/slos/SLODowntimePage/components/DowntimeList/DowntimeList.utils.ts:43]: use the 'Minutes' strings from constants

File: src/modules/85-cv/pages/slos/SLODowntimePage/components/DowntimeList/DowntimeList.utils.ts [hunk 7]
```diff
@@ -5,27 +5,91 @@
...
+    downtimeType = getString('common.occurrence.oneTime').toUpperCase()
+  } else {
+    const { downtimeDuration, downtimeRecurrence, recurrenceEndTime } = downtime?.spec?.spec as RecurringDowntimeSpec
+
+    timeFrame = `${getString('cv.sloDowntime.timeFrame', {
+      recurrenceType: getRecurrenceType(downtimeRecurrence, getString),
+      time: getFormattedTime({
+        time: startTime,
+        timezone,
+        format: 'LT'
```
Comment (harshilbits) [inline src/modules/85-cv/pages/slos/SLODowntimePage/components/DowntimeList/DowntimeList.utils.ts:67]: move this to constants.

File: src/modules/85-cv/pages/slos/SLODowntimePage/components/DowntimeList/DowntimeList.utils.ts [hunk 8]
```diff
@@ -5,27 +5,91 @@
...
+      }),
+      duration: getDuration(getString, downtimeDuration)
+    })} (${timezone})`
+    downtimeType = getString('cv.sloDowntime.recurringDowntime', {
+      startTime: getFormattedTime({
+        time: startTime,
+        timezone,
+        format: 'll'
```
Comment (harshilbits) [inline src/modules/85-cv/pages/slos/SLODowntimePage/components/DowntimeList/DowntimeList.utils.ts:75]: same.

File: src/modules/85-cv/pages/slos/components/CVCreateDowntime/CVCreateDowntime.utils.ts [hunk 9]
```diff
@@ -93,12 +100,44 @@ export const getDowntimeInitialFormData = (sloDowntime?: DowntimeDTO): DowntimeF
       time: recurringDowntimeSpec?.recurrenceEndTime,
       timezone: sloDowntime?.spec?.spec?.timezone
     }),
-    [DowntimeFormFields.ENTITIES_RULE_TYPE]: sloDowntime?.entitiesRule.type || 'Identifiers',
+    [DowntimeFormFields.ENTITIES_RULE_TYPE]: sloDowntime?.entitiesRule.type || EntitiesRuleType.IDENTIFIERS,
     [DowntimeFormFields.MS_LIST]: []
   }
+
+  if (type === DowntimeWindowToggleViews.ONE_TIME) {
```
Comment (harshilbits) [inline src/modules/85-cv/pages/slos/components/CVCreateDowntime/CVCreateDowntime.utils.ts:107]: this can be moved to separate function.

**Pull request merged**